stdlib/asm/str/rlp.masm

```diff
@@ -0,0 +1,128 @@
+# Total Number of VM Cycles: 638 #
+# This is used for 5 field-elements -> 4 field-elements RLP converting. #
+# Used to decode a RLP code which contains up to 34 chars, the decoding result is ASCII #
+# input: [a, b_0, b_1, b_2, b_3, b_4] #
+#   a: a_hi -- whether the decoding result will be put into memory #
+#      a_lo -- only if a_hi is '1', this is the memory address to put the decoding result in  #
+#   b_0 to b_4 are the RLP code #
+# output: a, b, (C)
+#   a: the number of chars in the RLP Code #
+#   b: b_hi -- whether the decoding result will be put into memory #
+#      b_lo -- only if b_hi is '1', this is the memory address to put the decoding result in  #
+#   C: the decoding result if not stored into memory. If the decoding result is stored into memory, C doesn't exist on stack. #
+
+export.rlp_decode.2
+    push.adv.1
+    pop.local.1                                                     #pop the first one into local[1]# 
+    # handle the first field element of RLP Code, which is unique to 2-5 field element.#
+    push.adv.1
+    u32split
+    swap
+    dup.0 push.255 u32and 
+    dup.0 push.162 lte assert                                       #make sure the string length is less than(or equal to) 34, otherwise error #
+    u32checked_sub.128 pop.local.0 u32shr.8                                    #local[0] should hold the first R-bytes, which represent the length of this string#
+    dup.0 dup.0 push.128 u32and not assert push.255 u32and swap     #the actuall 1st R-bytes, we need to make sure each char can be represent as ASCII#
+    u32shr.8
```

Comment:
I think what this sequence of instructions does is: remove 8 least significant bits from the element at the top of the stack, and make sure the resulting value is less than or equal to 128. If this is correct, it might be simpler to accomplish as follows:

1. First, we split a field element into two 32-bit chinks using `u32split` instruction.
2. Then, we can use `u32unchecked_divmod.256` to extract the least significant byte from a 32-bit element.
3. The, we can use `u32unchecked_divmod.128` to check if the 8th bit is set.

The sequence (assuming we've already done a `u32split`) would look something like this (I didn't actually test it, so it could have errors):
```
u32unchecked_divmod.256
u32unchecked_divmod.128
swap
eq.0
assert
```